src/linux_mcp_server/tools/system_info.py

```diff
@@ -350,156 +511,272 @@ async def get_memory_information(
+    external utilities like lspci or lsusb which may not be installed.
     """
-    try:
-        info = []
-        info.append("=== Hardware Information ===\n")
-
-        # Try lscpu for CPU info
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["lscpu"],
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("=== CPU Architecture (lscpu) ===")
-                info.append(stdout)
-        except FileNotFoundError:
-            info.append("CPU info: lscpu command not available")
-
-        # Try lspci for PCI devices
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["lspci"],
-                host=host,
-            )
-
-            if returncode == 0:
-                pci_lines = stdout.strip().split("\n")
-
-                info.append("\n=== PCI Devices ===")
-                # Show first 50 devices to avoid overwhelming output
-                for line in pci_lines[:50]:
-                    info.append(line)
-
-                if len(pci_lines) > 50:
-                    info.append(f"\n... and {len(pci_lines) - 50} more PCI devices")
-        except FileNotFoundError:
-            info.append("\nPCI devices: lspci command not available")
-
-        # Try lsusb for USB devices
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["lsusb"],
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("\n\n=== USB Devices ===")
-                info.append(stdout)
-        except FileNotFoundError:
-            info.append("\nUSB devices: lsusb command not available")
-
-        # Memory hardware info from dmidecode (requires root)
-        try:
-            returncode, stdout, stderr = await execute_command(
-                ["dmidecode", "-t", "memory"],
-                host=host,
-            )
-
-            if returncode == 0:
-                info.append("\n\n=== Memory Hardware (dmidecode) ===")
-                info.append(stdout)
-            elif "Permission denied" in stderr:
-                info.append("\n\nMemory hardware info: Requires root privileges (dmidecode)")
-        except FileNotFoundError:
-            info.append("\nMemory hardware info: dmidecode command not available")
-
-        if len(info) == 1:  # Only the header
-            info.append("No hardware information tools available.")
-
-        return "\n".join(info)
-    except Exception as e:
-        return f"Error getting hardware information: {str(e)}"
+
+    # Create a custom filter function that wraps filter_device_information
+    # to match the FilterFunc signature
+    async def custom_filter(parsed_data: ParsedData, fields: list[str] | None) -> ParsedData:
+        return await _filter_device_information(parsed_data, device_types=device_types, limit=limit)
```

Comment:
That this local function needs to exists seems wrong. This seems like `_filter_device_information` is not adhering to the interface contract for filter functions and needs to be changed instead of adding this translation layer.